sktime/clustering/k_means/k_means_sklearn.py

```diff
@@ -0,0 +1,264 @@
+    """
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+        "capability:missing_values": True,
+        "X_inner_mtype": ["pd-multiindex", "numpy3D"],
+        # required by the update_data utility
+        # otherwise, we could pass through to the distance directly
+    }
+
+    DELEGATED_PARAMS = [
+        "n_clusters",
+        "init",
+        "n_init",
+        "max_iter",
+        "tol",
+        "verbose",
+        "random_state",
+        "copy_x",
+        "algorithm",
+    ]
+    DELEGATED_FITTED_PARAMS = ["cluster_centers_", "labels_"]
+
+    def __init__(
+        self,
+        distance,
+        n_clusters=8,
+        init="k-means++",
+        n_init="auto",
+        max_iter=300,
+        tol=1e-4,
+        verbose=0,
+        random_state=None,
+        copy_x=True,
+        algorithm="lloyd",
+    ):
+        self.distance = distance
+        self.init = init
+        self.n_init = n_init
+        self.max_iter = max_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.random_state = random_state
+        self.copy_x = copy_x
+        self.algorithm = algorithm
+
+        super().__init__(n_clusters)
+
+        if isinstance(distance, BasePairwiseTransformerPanel):
+            tags_to_clone = [
+                "capability:multivariate",
+                "capability:unequal_length",
+                "capability:missing_values",
+            ]
+            self.clone_tags(distance, tags_to_clone)
+
+        self.kmeans_ = None
+
+    def _fit(self, X, y=None):
+        """Fit time series clusterer to training data.
+
+        Parameters
+        ----------
+        X : panel of time series, any sklearn Panel mtype
+            Time series to fit clusters to
+        y: ignored, exists for API consistency reasons
+
+        Returns
+        -------
+        self:
+            Fitted estimator.
+        """
+        self._X = X
+
+        distance = self.distance
+        distmat = distance(X)
+
+        deleg_param_dict = {key: getattr(self, key) for key in self.DELEGATED_PARAMS}
+
+        self.kmeans_ = KMeans(**deleg_param_dict)
+        self.kmeans_.fit(X=distmat)
+
+        for key in self.DELEGATED_FITTED_PARAMS:
+            if hasattr(self.kmeans_, key):
+                setattr(self, key, getattr(self.kmeans_, key))
+
+        return self
+
+    def _predict(self, X, y=None):
+        """Predict the closest cluster each sample in X belongs to.
+
+        Parameters
+        ----------
+        X : panel of time series, any sklearn Panel mtype
+            Time series instances to predict cluster indexes for
+        y: ignored, exists for API consistency reasons
+
+        Returns
+        -------
+        np.ndarray (1d array of shape (n_instances,))
+            Index of the cluster each time series in X belongs to
+        """
+        # if X is the same as seen in _fit, simply return the labels
+        if X is self._X:
```

Comment:
I think we don't need to do this, because the sklern estimator can handle different `X` in `predict`

Reply:
Thanks for starting the review. Addressed in commit bfd0be0. Let me know if that change is what you are looking for.

Reply:
yes, exactly!

The DBSCAN logic was just a crutch, since `sklearn` apparently allows estimators to fail in `predict` - we don't...